web/server/web_client.c

```diff
@@ -1064,7 +1064,7 @@ static inline HTTP_VALIDATION http_request_validate(struct web_client *w) {
                 // copy the URL - we are going to overwrite parts of it
                 // TODO -- ideally we we should avoid copying buffers around
                 strncpyz(w->last_url, w->decoded_url, NETDATA_WEB_REQUEST_URL_SIZE);
-                if (w->separator) {
+                if (w->separator && w->url_search_path) {
```

Comment:
I don't see how this will fix the corruption, the problem is not w->url_search_path being NULL. If it was NULL it would crash immediately. The problem is that it is not NULL and it goes and writes someplace where it shouldn't. Maybe it should be initialized to NULL at some other place as well?

Reply:
The same thing described above!